api/controllers/error_test.go

```diff
@@ -0,0 +1,91 @@
+package controllers
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/quintilesims/layer0/common/errors"
+	"github.com/quintilesims/layer0/common/models"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestErrorHandler(t *testing.T) {
+	cases := []struct {
+		Name           string
+		Error          error
+		ExpectedStatus int
+		ExpectedCode   errors.ErrorCode
+	}{
+		{
+			Name:           "Basic string error",
+			Error:          fmt.Errorf("some error"),
+			ExpectedStatus: http.StatusInternalServerError,
+			ExpectedCode:   errors.UnexpectedError,
+		},
+		{
+			Name:           "Unexpected ServerError",
+			Error:          errors.Newf(errors.UnexpectedError, ""),
+			ExpectedStatus: http.StatusInternalServerError,
+			ExpectedCode:   errors.UnexpectedError,
+		},
+		{
+			Name:           "ServerError: InvalidRequest",
+			Error:          errors.Newf(errors.InvalidRequest, ""),
+			ExpectedStatus: http.StatusBadRequest,
+			ExpectedCode:   errors.InvalidRequest,
+		},
+		{
+			Name:           "ServerError: DeployDoesNotExist",
+			Error:          errors.Newf(errors.DeployDoesNotExist, ""),
+			ExpectedStatus: http.StatusBadRequest,
```

Comment:
[nit] I would expect a `http.StatusNotFound` here.

Reply:
I went w/ StatusBadRequest because we can hit something like EnvironmentDoesNotExist on service create.